### PR TITLE
Make pages writable before actually writing on Windows

### PR DESCRIPTION
**Pull request by a20012251**

### CODE REVIEW

File: Headers/DebugServer2/Host/Windows/ExtraWrappers.h [hunk 1]
```diff
@@ -31,6 +31,13 @@ extern "C" {
 
 #define UNLEN 256
 
+WINBASEAPI SIZE_T WINAPI VirtualQueryEx(
```
Comment (sas) [inline Headers/DebugServer2/Host/Windows/ExtraWrappers.h]: Not a functional change, but can you put this next to `VirtualAllocEx` and friends?

File: Sources/Target/Windows/Process.cpp [hunk 2]
```diff
@@ -44,6 +44,12 @@ static uint8_t const gDebugBreakCode[] = {
 static const int gRtlExitUserThreadOffset = 0x02;
 #endif
 
+static DWORD getPageSize() {
```
Comment (sas) [inline Sources/Target/Windows/Process.cpp]: I think if you rebase on top of master you can get rid of this and just use `Platform::GetPageSize()`.

File: Headers/DebugServer2/Target/Windows/Process.h [hunk 3]
```diff
@@ -52,6 +52,16 @@ class Process : public ds2::Target::ProcessBase,
...
                         size_t *nwritten = nullptr) override;
 
+private:
+  ErrorCode makeMemoryWritable(Address const &address, size_t length,
+                               std::vector<MemoryRegionInfo> &modifiedRegions);
+  ErrorCode
+  restoreRegionsProtection(const std::vector<MemoryRegionInfo> &regions);
+  ErrorCode getMemoryRegionInfoInternal(Address const &address,
+                                        MemoryRegionInfo &info);
+  DWORD convertMemoryProtectionFromLLDB(uint32_t protection);
```
Comment (sas) [inline Headers/DebugServer2/Target/Windows/Process.h]: These are not converting to/from LLDB memory protection types, they're converting from ds2 to windows type and vice versa, so you should just call them to/from Windows I think. There are other functions like these for file mode conversions in one of @fjricci's patches. Look at how he did that maybe?

Comment (a20012251) [inline Headers/DebugServer2/Target/Windows/Process.h]: the naming makes sense, the other conversions were done inside a bigger function

**Comment by a20012251** [comment]
the travis failure seems to be unrelated

**Comment by fjricci** [comment]
Yeah, I broke stuff, you should rebase :)

**Comment by sas** [comment]
Committed as d9ffba46547850f4d806b59074ac3ab2a2d2f4e7.

**Pull request closed without merging**